### PR TITLE
[DOCS] Adds note about escaping backslashes in regex

### PR DESCRIPTION
**Pull request by abdonpijpelink**
The backslash character is an escape character in both JSON strings and regular expressions. This PR adds a note with an example about the need to escape both.

You can preview the change here: https://elasticsearch_89276.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/regexp-syntax.html#regexp-reserved-characters

### CODE REVIEW

**Review by immavalls — APPROVED** [review]
Thanks, @abdonpijpelink, for the fast response. It looks good to me, @crisdarocha might have more feedback as he's been involved in the investigation that led to this

**Review by crisdarocha — APPROVED** [review]
I like the change. Example is clear.

Would love to hear what Joe and Tim think about the topic. If this is enough, or if we need to be even more explicit.

File: docs/reference/query-dsl/regexp-syntax.asciidoc [hunk 1]
```diff
@@ -37,7 +37,55 @@ backslash or surround it with double quotes. For example:
...
+
+The backslash character is an escape character in both JSON strings and regular
+expressions. You need to escape both backslashes in a query, unless you use a
+language client, which takes care of this. For example, the keyword string `a\b`
+needs to be indexed as `"a\\b"`:
+
+////
+[source,console]
+----
+PUT my-index-000001
```
Comment (joegallo) [inline docs/reference/query-dsl/regexp-syntax.asciidoc:52]: I'm not sure the mapping here should be hidden -- if you try to reproduce this without using the same mapping (i.e. with `my_field` either implicitly or explicitly as `"type": "text"`) then the example doesn't quite work.

I defer to others, though, this is outside my wheelhouse.

Comment (abdonpijpelink) [reply docs/reference/query-dsl/regexp-syntax.asciidoc]: @joegallo I like your suggestion to simplify the example by removing the mapping and querying `my_field.keyword` instead. I've updated the example accordingly.

**Review by debadair — APPROVED** [review]
LGTM. Thanks for adding this!

**Pull request merged**